and-bible/app/src/main/java/net/bible/android/control/bookmark/BookmarkControl.java

```diff
@@ -67,6 +72,26 @@ public BookmarkControl(SwordContentFacade swordContentFacade, ActiveWindowPageMa
 		LABEL_UNLABELLED = new LabelDto(-998L, resourceProvider.getString(R.string.label_unlabelled), null);
 	}
 
+	public void updateBookmarkSettings(PlaybackSettings settings) {

```

Comment:
1. We need to figure out a way to make it more obvious or keep it ‘less obvious’ (it is not enabled by default anyway) and possibly even hide it from basic users. These are the solutions that come to my mind:
    - Improve help screen text and try to provide more UI hints so that functionality is better understood. Perhaps “Speak settings” (that can be optionally saved in bookmark) could be more clearly separated from other settings on that activity. I can try to think about this more later.
    - Hide the feature from basic users (see answer to 2.)
2. Default is disabled, so I don’t think this is such a big problem. Also if we manage to make it ‘more obvious’ to users, maybe it is not a problem any more then. If it stays less obvious despite our efforts, we could then consider creating “Enable advanced features” option  in main settings window, and show this particular “Save speak settings in bookmarks” option only if “advanced features” option is enabled. I.e. hide this from “ordinary users” (keep it disabled). Similarly as we already now do for “Manually installed documents exists” option.
3. I don’t think this is a problem at all. This is design choice, bookmark data could contain other application related metadata if it is beneficial somehow (like it is here). Extra columns in sqlite database are cheap anyway.
4. I don’t know but I don’t think this is obvious. I think there might be many other ‘advanced’ users that find this feature very useful. I could provide youtube help video(s)/tutorials for users so that they adopt this feature more.
5. For me personally removing the feature would not help at all. I would need to maintain the code anyway ‘forever’ in my own fork. This code is very carefully tested via automated tests so I don't think keeping this is such a big problem.

Due the nature of this feature, it has code scattered in quite many places (if I remember correctly) and also your requirement of removing the database field would make maintaining my own branch troublesome. So I don't think removing this feature is an acceptable solution to me. We need to find / negotiate another solution.

Reply:
As you say, it is scattered around the code and does not provide enough benefit and will confuse users.  We just need to remove it.

Reply:
I will try to make this as obvious and non-confusing as possible. I will think about this more later, but I'm marking this discussion now as resolved, as I'm not planning to remove this feature.